src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -165,22 +185,77 @@ class TaskDeploymentService {
     }
 
     /**
-     * Helper method to initialise the lateinit vars [taskAssumeRoleString] and [taskRolePolicyString] by
-     * converting the associated `@Value` parameters to Strings and replacing `ADDITIONAL_PERMISSIONS` in
-     * [taskRolePolicyString] with the provided [additionalPermissions]
+     * Helper method to initialise the lateinit vars [taskRolePolicyString] and [jupyterBucketAccessRolePolicyString] by
+     * converting the JSON to an instance of `JsonObject` data class and adding any requirements before serialising back to JSON
      *
-     * @return [Pair] of [taskRolePolicyString] to [taskAssumeRoleString] for ease of access.
+     * @return [taskRolePolicyString] and [jupyterBucketAccessRolePolicyString] respectively.
      */
-    fun parsePolicyDocuments(additionalPermissions: List<String>): Pair<String, String> {
-        var replaceString = ""
-        if (additionalPermissions.isNotEmpty()) {
-            logger.info("Adding permissions to containers", "permissions" to additionalPermissions.joinToString())
-            val permissionsJson = additionalPermissions.joinToString(prefix = "\"", separator = "\",\"", postfix = "\"")
-            replaceString = "$permissionsJson,"
+    fun parsePolicyDocument(resource: Resource, sidAndAdditions: Map<String, List<String>>, key: String): String{
+        val mapper = ObjectMapper()
+                .setPropertyNamingStrategy(AwsPropertyNamingStrategy())
+        val obj = mapper.readValue(resource.url, AwsIamPolicyJsonObject::class.java)
+        obj.Statement.forEach {statement ->
+            sidAndAdditions.forEach {
+                if(it.key == statement.Sid) {
+                    if (key == "Resource") statement.Resource = statement.Resource.plus(it.value)
+                    else if (key == "Action") statement.Action = statement.Action.plus(it.value)
+                    else throw IllegalArgumentException("Key does not match expected values: \"Resource\" or \"Action\"")
+                }
+            }
+        }
+        return mapper.writeValueAsString(obj)
+    }
```

Comment:
At first glance it is quite difficult to understand what this method does. I suggest refactoring to something like 
```suggestion
    fun parsePolicyDocument(resource: Resource, additionsForSid: Map<String, List<String>>, statementKeyToUpdate: String): String{
        val mapper = ObjectMapper()
                .setPropertyNamingStrategy(CustomPropertyNamingStrategy())
        val obj = mapper.readValue(resource.url, JsonObject::class.java)

        val updatedStatements = obj.Statement.map { statement ->
            val additions = additionsForSid[statement.Sid] ?: return@map statement
            when(statementKeyToUpdate) {
                "Resource" -> return@map statement.let { it.Resource = it.Resource.plus(additions); it }
                "Action" -> return@map statement.let { it.Action = it.Action.plus(additions); it }
                else -> throw IllegalArgumentException("statementKeyToUpdate must be one of \"Resource\" or \"Action\"")
            }
        }

        val updatedObject = obj.let { it.Statement = updatedStatements; it }
        return mapper.writeValueAsString(updatedObject)
    }
```

Always try to avoid mutating objects and instead create new ones and return them, it's more clear and it guarantees no side effects